libvips/foreign/vips2webp.c

```diff
@@ -82,6 +83,73 @@ get_preset( VipsForeignWebpPreset preset )
 	return( -1 );
 }
 
+#if WEBP_ENCODER_ABI_VERSION < 0x0100
+typedef struct {
+	uint8_t *mem;
+	size_t size;
+	size_t max_size;
+	uint32_t pad[1];
+} WebPMemoryWriter;
+
+static void WebPMemoryWriterInit( WebPMemoryWriter *writer ) {
+	writer->mem = NULL;
+	writer->size = 0;
+	writer->max_size = 0;
+}
+
+static int WebPMemoryWrite( const uint8_t *data, size_t data_size,
+	const WebPPicture *picture ) {
+	WebPMemoryWriter* const w = (WebPMemoryWriter*) picture->custom_ptr;
+	size_t next_size;
+
+	if (w == NULL) {
+		return 1;
```

Comment:
This is copied from libwebp-0.1.99, but is essentially the same in 0.5.0. I chose the older version because it does not use libwebp memory allocation functions that are missing in 0.1.3.